test/ui/scopeQuery.spec.js

```diff
@@ -0,0 +1,81 @@
+const scopeQuery = require('../../examples/lib/scopeQuery');
+
+describe('scopeQuery Method', function() {
+
+  beforeAll(() => {
+    const { JSDOM } = require('jsdom');
+
+    var { window } = new JSDOM(`
+      <body>
+        <div id="scope1">
+          <div class="child" id="child1">Child1</div>
+          <div class="child" id="child2">Child2</div>
+          <div class="child" id="child3">Child3</div>
+          <div class="child" id="child4">Child4</div>
+          <div class="child" id="child5">Child5</div>
+          <div class="child" id="child6">Child6</div>
+        </div>
+
+        <div id="scope2">
+          <div class="child" id="child1">Child1Scope2</div>
+          <div class="child" id="child2">Child2Scope2</div>
+          <div class="child" id="child3">Child3Scope2</div>
+          <div class="child" id="child4">Child4Scope2</div>
+          <div class="child" id="child5">Child5Scope2</div>
+        </div>
+      </body>`);
+
+    const { document } = window;
+    global.window = window;
+    global.document = document;
+    const $ = global.jQuery = global.$ = require('jquery');
+  })
+  
+  
+  it('scope 1 querySelector', function() {
+
+    const $scope1 = scopeQuery.scopeSelector(document.querySelector('#scope1'))
+    const $scope1All = scopeQuery.scopeSelectorAll(document.querySelector('#scope1'))
+
+    expect($scope1('.child').length).toBe(1);
+    expect($scope1All('.child').length).toBe(6);
+
+    expect($scope1('#child1').text()).toBe('Child1');
+    expect($scope1('#child2').text()).toBe('Child2');
+
+    expect($scope1All('#child1').text()).toBe('Child1');
+    expect($scope1All('#child2').text()).toBe('Child2');
+
+    expect($scope1().elem('.child').length).toBe(1);
+    expect($scope1().elemAll('.child').length).toBe(6);
+
+    expect($scope1All().elem('.child').length).toBe(1);
+    expect($scope1All().elemAll('.child').length).toBe(6);
+  })
+
+  it('scope 2 querySelector', function() {
+
+    const $scope2 = scopeQuery.scopeSelector(document.querySelector('#scope2'))
+    const $scope2All = scopeQuery.scopeSelectorAll(document.querySelector('#scope2'))
+
+    expect($scope2('.child').length).toBe(1);
+    expect($scope2All('.child').length).toBe(5);
+
+    expect($scope2('#child1').text()).toBe('Child1Scope2');
+    expect($scope2('#child2').text()).toBe('Child2Scope2');
+
+    expect($scope2All('#child1').text()).toBe('Child1Scope2');
+    expect($scope2All('#child2').text()).toBe('Child2Scope2');
+
+    expect($scope2().elem('.child').length).toBe(1);
+    expect($scope2().elemAll('.child').length).toBe(5);
+
+    expect($scope2All().elem('.child').length).toBe(1);
+    expect($scope2All().elemAll('.child').length).toBe(5);
+  })
+
+})
+
```

Comment:
Okay so basically you are testing your new method by creating an example html! Why don't we test it for a particular step? I mean a module. I understand the results would be same but I want to know if you tried doing that and have been stuck as it happened with me as well!

Reply:
It took me a day to figure this out.

Reply:
Yeah the way is great as we basically have to test the functionality but have you tried it for a module? I just want to know if you faced issues with doing it with jasmine-jquery and fixtures and spyOnEvent

Reply:
@aashna27 I didn't understand your question properly. If there is some problem in my code, please let me know here. If you want to ask some questions, DM me on gitter and resolve this conversation. Thanks!

Reply:
I hear what @aashna27 is saying here, but although I see the logic in the suggestion of the HTML fixture being of similar structure to the one @HarshKhandeparkar has used, I think it's basically OK for now. The important thing is that it serves a generalized purpose and that the HTML example here properly demonstrates it. Thanks!!!

Reply:
Ok, now I understand. This is just a unit test, maybe we can write integration tests in the future. Currently, even the unit tests are incomplete so maybe integration tests and e2e tests can be written in the near future. What do ya'll say?

Reply:
> @aashna27 I didn't understand your question properly. If there is some problem in my code, please let me know here. If you want to ask some questions, DM me on gitter and resolve this conversation. Thanks!

No no problem with the code just wanted to know if you tried another approach, this one is great though!